Quarkus based images do not work with BoringSSL

The instructions given in [Using OpenSSL](https://www.eclipse.org/hono/docs/admin-guide/secure_communication/#using-openssl) do not work with the Quarkus based images.
When starting up a container that contains the BoringSSL jar file in `/opt/hono/extensions`, the shared library cannot be loaded:
```sh
HTTP2022-02-24 06:43:37,771 DEBUG [io.net.han.ssl.OpenSsl] (vert.x-eventloop-thread-2) Failed to load netty-tcnative; OpenSslEngine will be unavailable, unless the application has already loaded the symbols by some other means. See https://netty.io/wiki/forked-tomcat-native.html for more information.: java.lang.IllegalArgumentException: Failed to load any of the given libraries: [netty_tcnative_linux_x86_64, netty_tcnative_linux_x86_64_fedora, netty_tcnative_x86_64, netty_tcnative]
HTTP	at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:114)
HTTP	at io.netty.handler.ssl.OpenSsl.loadTcNative(OpenSsl.java:705)
HTTP	at io.netty.handler.ssl.OpenSsl.<clinit>(OpenSsl.java:146)
HTTP	at org.eclipse.hono.connection.impl.ConnectionFactoryImpl.addTlsTrustOptions(ConnectionFactoryImpl.java:383)
HTTP	at org.eclipse.hono.connection.impl.ConnectionFactoryImpl.addOptions(ConnectionFactoryImpl.java:360)
HTTP	at org.eclipse.hono.connection.impl.ConnectionFactoryImpl.connect(ConnectionFactoryImpl.java:144)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl$ConnectionAttempt.connect(HonoConnectionImpl.java:1096)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl$ConnectionAttempt.start(HonoConnectionImpl.java:1059)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.lambda$connect$2(HonoConnectionImpl.java:387)
HTTP	at org.eclipse.hono.util.HonoProtonHelper.executeOnContext(HonoProtonHelper.java:230)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.executeOnContext(HonoConnectionImpl.java:236)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.connect(HonoConnectionImpl.java:377)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.connect(HonoConnectionImpl.java:358)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.connect(HonoConnectionImpl.java:352)
HTTP	at org.eclipse.hono.client.amqp.AbstractServiceClient.start(AbstractServiceClient.java:274)
HTTP	at org.eclipse.hono.client.notification.amqp.ProtonBasedNotificationReceiver.start(ProtonBasedNotificationReceiver.java:77)
HTTP	at org.eclipse.hono.util.WrappedLifecycleComponentVerticle.start(WrappedLifecycleComponentVerticle.java:41)
HTTP	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$5(DeploymentManager.java:196)
HTTP	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
HTTP	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:63)
HTTP	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:38)
HTTP	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
HTTP	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
HTTP	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:387)
HTTP	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
HTTP	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
HTTP	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
HTTP	at java.base/java.lang.Thread.run(Unknown Source)
HTTP	Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_tcnative_linux_x86_64
HTTP		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:224)
HTTP		at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:105)
HTTP		... 27 more
HTTP	Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_tcnative_linux_x86_64.so
HTTP		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:171)
HTTP		... 28 more
HTTP		Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_linux_x86_64 in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
HTTP			at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source)
HTTP			at java.base/java.lang.Runtime.loadLibrary0(Unknown Source)
HTTP			at java.base/java.lang.System.loadLibrary(Unknown Source)
HTTP			at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
HTTP			at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:376)
HTTP			at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:146)
HTTP			... 28 more
HTTP			Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_linux_x86_64 in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
HTTP				at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source)
HTTP				at java.base/java.lang.Runtime.loadLibrary0(Unknown Source)
HTTP				at java.base/java.lang.System.loadLibrary(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
HTTP				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
HTTP				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
HTTP				at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
HTTP				at java.base/java.lang.reflect.Method.invoke(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:410)
HTTP				at java.base/java.security.AccessController.doPrivileged(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:402)
HTTP				at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:368)
HTTP				... 29 more
HTTP	Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_tcnative_linux_x86_64_fedora
HTTP		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:224)
HTTP		at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:105)
HTTP		... 27 more
HTTP	Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_tcnative_linux_x86_64_fedora.so
HTTP		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:171)
HTTP		... 28 more
HTTP		Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_linux_x86_64_fedora in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
HTTP			at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source)
HTTP			at java.base/java.lang.Runtime.loadLibrary0(Unknown Source)
HTTP			at java.base/java.lang.System.loadLibrary(Unknown Source)
HTTP			at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
HTTP			at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:376)
HTTP			at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:146)
HTTP			... 28 more
HTTP			Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_linux_x86_64_fedora in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
HTTP				at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source)
HTTP				at java.base/java.lang.Runtime.loadLibrary0(Unknown Source)
HTTP				at java.base/java.lang.System.loadLibrary(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
HTTP				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
HTTP				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
HTTP				at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
HTTP				at java.base/java.lang.reflect.Method.invoke(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:410)
HTTP				at java.base/java.security.AccessController.doPrivileged(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:402)
HTTP				at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:368)
HTTP				... 29 more
HTTP	Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_tcnative_x86_64
HTTP		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:224)
HTTP		at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:105)
HTTP		... 27 more
HTTP	Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_tcnative_x86_64.so
HTTP		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:171)
HTTP		... 28 more
HTTP		Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_x86_64 in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
HTTP			at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source)
HTTP			at java.base/java.lang.Runtime.loadLibrary0(Unknown Source)
HTTP			at java.base/java.lang.System.loadLibrary(Unknown Source)
HTTP			at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
HTTP			at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:376)
HTTP			at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:146)
HTTP			... 28 more
HTTP			Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative_x86_64 in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
HTTP				at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source)
HTTP				at java.base/java.lang.Runtime.loadLibrary0(Unknown Source)
HTTP				at java.base/java.lang.System.loadLibrary(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
HTTP				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
HTTP				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
HTTP				at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
HTTP				at java.base/java.lang.reflect.Method.invoke(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:410)
HTTP				at java.base/java.security.AccessController.doPrivileged(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:402)
HTTP				at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:368)
HTTP				... 29 more
HTTP	Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_tcnative
HTTP		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:224)
HTTP		at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:105)
HTTP		... 27 more
HTTP	Caused by: java.io.FileNotFoundException: META-INF/native/libnetty_tcnative.so
HTTP		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:171)
HTTP		... 28 more
HTTP		Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
HTTP			at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source)
HTTP			at java.base/java.lang.Runtime.loadLibrary0(Unknown Source)
HTTP			at java.base/java.lang.System.loadLibrary(Unknown Source)
HTTP			at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
HTTP			at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:376)
HTTP			at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:146)
HTTP			... 28 more
HTTP			Suppressed: java.lang.UnsatisfiedLinkError: no netty_tcnative in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
HTTP				at java.base/java.lang.ClassLoader.loadLibrary(Unknown Source)
HTTP				at java.base/java.lang.Runtime.loadLibrary0(Unknown Source)
HTTP				at java.base/java.lang.System.loadLibrary(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryUtil.loadLibrary(NativeLibraryUtil.java:38)
HTTP				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
HTTP				at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
HTTP				at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
HTTP				at java.base/java.lang.reflect.Method.invoke(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryLoader$1.run(NativeLibraryLoader.java:410)
HTTP				at java.base/java.security.AccessController.doPrivileged(Unknown Source)
HTTP				at io.netty.util.internal.NativeLibraryLoader.loadLibraryByHelper(NativeLibraryLoader.java:402)
HTTP				at io.netty.util.internal.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:368)
HTTP				... 29 more
HTTP
HTTP2022-02-24 06:43:37,775 DEBUG [io.net.han.ssl.OpenSsl] (vert.x-eventloop-thread-2) Initialize netty-tcnative using engine: 'default'
HTTP2022-02-24 06:43:37,777 DEBUG [io.net.han.ssl.OpenSsl] (vert.x-eventloop-thread-2) Failed to initialize netty-tcnative; OpenSslEngine will be unavailable. See https://netty.io/wiki/forked-tomcat-native.html for more information.: java.lang.UnsatisfiedLinkError: 'int io.netty.internal.tcnative.Library.aprMajorVersion()'
HTTP	at io.netty.internal.tcnative.Library.aprMajorVersion(Native Method)
HTTP	at io.netty.internal.tcnative.Library.initialize(Library.java:191)
HTTP	at io.netty.handler.ssl.OpenSsl.initializeTcNative(OpenSsl.java:710)
HTTP	at io.netty.handler.ssl.OpenSsl.<clinit>(OpenSsl.java:163)
HTTP	at org.eclipse.hono.connection.impl.ConnectionFactoryImpl.addTlsTrustOptions(ConnectionFactoryImpl.java:383)
HTTP	at org.eclipse.hono.connection.impl.ConnectionFactoryImpl.addOptions(ConnectionFactoryImpl.java:360)
HTTP	at org.eclipse.hono.connection.impl.ConnectionFactoryImpl.connect(ConnectionFactoryImpl.java:144)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl$ConnectionAttempt.connect(HonoConnectionImpl.java:1096)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl$ConnectionAttempt.start(HonoConnectionImpl.java:1059)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.lambda$connect$2(HonoConnectionImpl.java:387)
HTTP	at org.eclipse.hono.util.HonoProtonHelper.executeOnContext(HonoProtonHelper.java:230)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.executeOnContext(HonoConnectionImpl.java:236)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.connect(HonoConnectionImpl.java:377)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.connect(HonoConnectionImpl.java:358)
HTTP	at org.eclipse.hono.client.impl.HonoConnectionImpl.connect(HonoConnectionImpl.java:352)
HTTP	at org.eclipse.hono.client.amqp.AbstractServiceClient.start(AbstractServiceClient.java:274)
HTTP	at org.eclipse.hono.client.notification.amqp.ProtonBasedNotificationReceiver.start(ProtonBasedNotificationReceiver.java:77)
HTTP	at org.eclipse.hono.util.WrappedLifecycleComponentVerticle.start(WrappedLifecycleComponentVerticle.java:41)
HTTP	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$5(DeploymentManager.java:196)
HTTP	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
HTTP	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:63)
HTTP	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:38)
HTTP	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
HTTP	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
HTTP	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:387)
HTTP	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
HTTP	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
HTTP	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
HTTP	at java.base/java.lang.Thread.run(Unknown Source)
HTTP
```

--- a/adapters/base-quarkus/pom.xml
+++ b/adapters/base-quarkus/pom.xml
@@ -145,10 +145,12 @@
                       <env>
                         <JAVA_MAJOR_VERSION>${maven.compiler.release}</JAVA_MAJOR_VERSION>
                         <JAVA_APP_NAME>${project.artifactId}</JAVA_APP_NAME>
-                        <JAVA_LIB_DIR>/opt/hono/lib/*:/opt/hono/extensions/*</JAVA_LIB_DIR>
+                        <JAVA_MAIN_CLASS>io.quarkus.bootstrap.runner.QuarkusEntryPoint</JAVA_MAIN_CLASS>
+                        <JAVA_LIB_DIR>app/*:quarkus/*:lib/boot/*:lib/main/*:extensions/*</JAVA_LIB_DIR>
                         <!-- use Base64 encoder/decoder that is compatible with vert.x 3 -->
-                        <JAVA_OPTIONS>-Dvertx.json.base64=legacy -Djava.util.logging.manager=org.jboss.logmanager.LogManager</JAVA_OPTIONS>
-                        <JAVA_APP_JAR>quarkus-run.jar</JAVA_APP_JAR>
+                        <JAVA_OPTIONS>
+                          -Dvertx.json.base64=legacy
+                          -Djava.util.logging.manager=org.jboss.logmanager.LogManager</JAVA_OPTIONS>
                       </env>
                       <entryPoint>
                         <arg>/opt/hono/run-java.sh</arg>

--- a/services/base-quarkus/pom.xml
+++ b/services/base-quarkus/pom.xml
@@ -142,10 +142,13 @@
                       <env>
                         <JAVA_MAJOR_VERSION>${maven.compiler.release}</JAVA_MAJOR_VERSION>
                         <JAVA_APP_NAME>${project.artifactId}</JAVA_APP_NAME>
-                        <JAVA_LIB_DIR>/opt/hono/lib/*:/opt/hono/extensions/*</JAVA_LIB_DIR>
+                        <JAVA_MAIN_CLASS>io.quarkus.bootstrap.runner.QuarkusEntryPoint</JAVA_MAIN_CLASS>
+                        <JAVA_LIB_DIR>app/*:quarkus/*:lib/boot/*:lib/main/*:extensions/*</JAVA_LIB_DIR>
                         <!-- use Base64 encoder/decoder that is compatible with vert.x 3 -->
-                        <JAVA_OPTIONS>-Dvertx.json.base64=legacy -Djava.util.logging.manager=org.jboss.logmanager.LogManager</JAVA_OPTIONS>
-                        <JAVA_APP_JAR>quarkus-run.jar</JAVA_APP_JAR>
+                        <JAVA_OPTIONS>
+                          -Dvertx.json.base64=legacy
+                          -Djava.util.logging.manager=org.jboss.logmanager.LogManager
+                        </JAVA_OPTIONS>
                       </env>
                       <entryPoint>
                         <arg>/opt/hono/run-java.sh</arg>

--- a/site/documentation/content/admin-guide/secure_communication.md
+++ b/site/documentation/content/admin-guide/secure_communication.md
@@ -230,6 +230,11 @@ the `--mount` parameter. Please refer to the
 [Docker reference documentation](https://docs.docker.com/engine/reference/commandline/service_create/#add-bind-mounts-volumes-or-memory-filesystems)
 for details.
 
+{{% warning %}}
+Configuring containers to use OpenSSL the way described above does **not** work with container images based on
+native executables.
+{{% /warning %}}
+
 ## Server Name Indication (SNI)
 
 [Server Name Indication](https://tools.ietf.org/html/rfc6066#section-3) can be used to indicate to a server the host

--- a/site/homepage/content/release-notes.md
+++ b/site/homepage/content/release-notes.md
@@ -13,6 +13,8 @@ description = "Information about changes in recent Hono releases. Includes new f
   *hono.mongodb.connectionString* property. This has been fixed.
 * When a tenant or device gets disabled or deleted, any open AMQP or MQTT connections from clients having authenticated
   themselves as belonging to that tenant or device are getting closed now.
+* Using OpenSSL with the Quarkus based variant of Hono components did not work as described in the Secure Communication
+  guide. This has been fixed.
 
 ### API Changes
 

--- a/tests/pom.xml
+++ b/tests/pom.xml
@@ -168,7 +168,7 @@
     <hono.http-adapter.image>hono-adapter-http-vertx</hono.http-adapter.image>
     <hono.http-adapter.config-dir>opt/hono/config</hono.http-adapter.config-dir>
     <hono.http-adapter.max-mem>314572800</hono.http-adapter.max-mem>
-    <hono.http-adapter.nativeTlsRequired>false</hono.http-adapter.nativeTlsRequired>
+    <hono.http-adapter.nativeTlsRequired>true</hono.http-adapter.nativeTlsRequired>
     <hono.http-adapter.java-options>${default.java-options}</hono.http-adapter.java-options>
     <hono.http-adapter.native-image-args></hono.http-adapter.native-image-args>
 
@@ -184,6 +184,7 @@
     <hono.auth-server.max-mem>205520896</hono.auth-server.max-mem>
     <hono.auth-server.java-options>${default.java-options}</hono.auth-server.java-options>
     <hono.auth-server.native-image-args></hono.auth-server.native-image-args>
+    <hono.auth-server.nativeTlsRequired>true</hono.auth-server.nativeTlsRequired>
 
     <trace.frames>0</trace.frames>
     <jaeger.disabled>true</jaeger.disabled>
@@ -526,7 +527,6 @@
         <hono.amqp-adapter.config-dir>etc/hono</hono.amqp-adapter.config-dir>
         <hono.coap-adapter.config-dir>etc/hono</hono.coap-adapter.config-dir>
         <hono.http-adapter.config-dir>etc/hono</hono.http-adapter.config-dir>
-        <hono.http-adapter.nativeTlsRequired>true</hono.http-adapter.nativeTlsRequired>
         <hono.mqtt-adapter.config-dir>etc/hono</hono.mqtt-adapter.config-dir>
         <hono.auth-server.config-dir>etc/hono</hono.auth-server.config-dir>
         <hono.command-router.config-dir>etc/hono</hono.command-router.config-dir>
@@ -545,18 +545,16 @@
       <properties>
         <hono.image-suffix>-quarkus-native</hono.image-suffix>
         <hono.amqp-adapter.max-mem>67108864</hono.amqp-adapter.max-mem>
-        <hono.amqp-adapter.useNativeTransport>false</hono.amqp-adapter.useNativeTransport>
 
         <hono.coap-adapter.max-mem>67108864</hono.coap-adapter.max-mem>
-        <hono.coap-adapter.useNativeTransport>false</hono.coap-adapter.useNativeTransport>
 
         <hono.http-adapter.max-mem>67108864</hono.http-adapter.max-mem>
         <hono.http-adapter.nativeTlsRequired>false</hono.http-adapter.nativeTlsRequired>
 
         <hono.mqtt-adapter.max-mem>67108864</hono.mqtt-adapter.max-mem>
-        <hono.mqtt-adapter.useNativeTransport>false</hono.mqtt-adapter.useNativeTransport>
 
         <hono.auth-server.max-mem>33554432</hono.auth-server.max-mem>
+        <hono.auth-server.nativeTlsRequired>false</hono.auth-server.nativeTlsRequired>
         <hono.command-router.max-mem>67108864</hono.command-router.max-mem>
         <hono.deviceregistry.mongodb.max-mem>150000000</hono.deviceregistry.mongodb.max-mem>
         <hono.deviceregistry.image-suffix>-quarkus</hono.deviceregistry.image-suffix>
@@ -747,6 +745,15 @@
                       <basedir>/</basedir>
                       <inline>
                         <id>config</id>
+                        <dependencySets>
+                          <dependencySet>
+                            <includes>
+                              <!--  use openSSL in Auth server -->
+                              <include>io.netty:netty-tcnative-boringssl-static</include>
+                            </includes>
+                            <outputDirectory>opt/hono/extensions</outputDirectory>
+                          </dependencySet>
+                        </dependencySets>
                         <fileSets>
                           <fileSet>
                             <directory>${project.build.directory}/resources/auth</directory>

--- a/tests/src/test/resources/auth/application.yml
+++ b/tests/src/test/resources/auth/application.yml
@@ -9,6 +9,7 @@ hono:
       bindAddress: "0.0.0.0"
       keyPath: /etc/hono/certs/auth-server-key.pem
       certPath: /etc/hono/certs/auth-server-cert.pem
+      nativeTlsRequired: ${hono.auth-server.nativeTlsRequired}
     svc:
       permissionsPath: "/${hono.auth-server.config-dir}/permissions.json"
       signing:

